test/test_torch.py

```diff
@@ -12340,6 +12340,172 @@ def test_against_np(tensor, bins=100, min=0, max=0):
             expanded = torch.randn(1, 5, 1, 2, device=device).expand(3, 5, 7, 2)
             test_against_np(expanded)
 
+    @onlyCUDA
+    @expectedAlertNondeterministic('_histogram_cuda', fn_has_device_arg=False)
+    def test_histogram_alert_nondeterministic(self, device):
+        torch.histogram(torch.tensor([], device=device))
+
+    @onlyCPU
+    def test_histogram_bin_edges_sorted(self, device):
+        # bins not sorted - the check is only performed on the CPU to avoid device synchronization
+        with self.assertRaisesRegex(RuntimeError, 'bin edges must increase monotonically'):
+            torch.histogram(torch.tensor([1], dtype=torch.float, device=device),
+                            bins=torch.tensor([2, 1], dtype=torch.float, device=device))
+
+    def test_histogram(self, device):
```

Comment:
This test is very good. Maybe I missed this, but is there a test that simultaneously setting bins with a tensor and providing a range throws a runtime error?

This is allowed in NumPy's histogram function but is extremely misleading since the value given to ranges is ignored in this case, so I'd prefer we throw an error, instead.

Reply:
There is a test for that now.